lib/firefox/firefoxBidi.js

```diff
@@ -146,4 +147,54 @@ export class FirefoxBidi {
       }
     }
   }
+
+  async setRequestHeaders(requestHeaders) {
+    const headersArray = toArray(requestHeaders);
+    const headers = [];
+    for (let header of headersArray) {
+      if (header.indexOf && header.includes(':')) {
+        const parts = header.split(':');
+        headers.push({
+          name: parts[0],
+          value: {
+            type: 'string',
+            value: parts[1]
+          }
+        });
+      } else {
+        log.error(
+          'Request headers need to be of the format key:value not ' + header
+        );
+      }
+    }
+
+    const command = {
+      method: 'network.addIntercept',
+      params: {
+        phases: ['beforeRequestSent']
+      }
+    };
+    const bidi = new Bidi(this.driver, this.options.browser);
+
+    await bidi.send(command);
+    await bidi.subscribe('network.beforeRequestSent');
+    await bidi.onMessage(async function (event) {
+      const parsedEvent = JSON.parse(Buffer.from(event.toString()));
+      if (parsedEvent.method === 'network.beforeRequestSent') {
+        const continueRequest = {
+          method: 'network.continueRequest',
+          params: {
+            request: parsedEvent.params.request.request,
+            headers
```

Comment:
Mentioned it on bugzilla, but here you should merge your headers with the current request headers.
```suggestion
            [headers](headers: [...parsedEvent.params.request.headers, ...headers],)
```

Reply:
Ah thank you! I got confused since I got "success" without the rest of the headers.